asllib/aslspec/latex.ml

```diff
@@ -0,0 +1,267 @@
+let pp_mathhypertarget fmt target_str =
+  fprintf fmt {|\mathhypertarget{%s}|} target_str
+
+(** A type enumerating different kinds of parentheses. *)
+type parenthesis = Parens | Braces | Brackets
+
+(** [pp_parenthesized parenthesis large pp_elem fmt elem] formats [elem] with
+    the formatter [fmt] using [pp_elem] and surrounds it with the specified
+    [parenthesis]. If [large] is true, then large versions of the parentheses
+    are used. *)
+let pp_parenthesized parenthesis large pp_elem fmt elem =
+  let left = if large then {|\left|} else "" in
+  let right = if large then {|\right|} else "" in
+  match parenthesis with
+  | Parens -> fprintf fmt "%s(%a%s)" left pp_elem elem right
+  | Braces -> fprintf fmt "%s\\{%a%s\\}" left pp_elem elem right
+  | Brackets -> fprintf fmt "%s[%a%s]" left pp_elem elem right
+
+(** [over_text fmt fmt_top top fmt_bottom bottom] renders
+    [\overtext{top}{bottom}] where [top] and [bottom] are formatted using
+    [fmt_top] and [fmt_bottom] respectively. *)
+let over_text fmt fmt_top top fmt_bottom bottom =
+  fprintf fmt {|\overtext{%a}{%a}|} fmt_top top fmt_bottom bottom
+
+(** [pp_field_name_to_latex field_name] formats [field_name] - a field name
+    appearing in the specification text - in a form suitable for LaTeX. *)
+let field_name_to_latex field_name =
+  spec_var_to_latex_var ~font_type:Text field_name
+
+(** [pp_var fmt var_str] formats [var_str] - a variable appearing in the
+    specification text - in a form suitable for LaTeX. *)
+let pp_var fmt var_str =
+  pp_print_string fmt (spec_var_to_latex_var ~font_type:TextTT var_str)
+
+(** [pp_latex_array alignment fmt pp_fun_rows] renders a table of elements using
+    a LaTeX array environment. The [alignment] string specifies the alignment of
+    each column and is copied directly to the array environment. The
+    [pp_fun_rows] is a list of lists, where the outer list represents the array
+    rows and the inner lists represent the columns. *)
+let pp_latex_array alignment fmt pp_fun_rows =
+  let () = assert (Str.string_match (Str.regexp "[lcr]+$") alignment 0) in
+  let num_columns = String.length alignment in
+  let pp_elt = fun fmt pp_fun -> pp_fun fmt in
+  let pp_one_row fmt pp_funs =
+    let () = assert (List.compare_length_with pp_funs num_columns = 0) in
+    pp_print_list ~pp_sep:(fun fmt () -> fprintf fmt " & ") pp_elt fmt pp_funs
+  in
+  fprintf fmt "@[<v>\\begin{array}{%s}@ %a@ \\end{array}@]" alignment
+    (pp_print_list ~pp_sep:(fun fmt () -> fprintf fmt {|\\@ |}) pp_one_row)
+    pp_fun_rows
+
+(** [pp_sep_elements ~pp_sep ~alignment pp_elem layout fmt elements] formats a
+    list of [elements] separated by [pp_sep] according to [layout]. Each element
+    is formatted using [pp_elem]. The [alignment] string specifies the alignment
+    of each column and is copied directly to the array environment when the
+    [layout] is vertical. *)
+let pp_sep_elements ~pp_sep ~alignment pp_elem layout fmt elements =
```

Comment:
`pp_sep_elements` and `pp_connect_elements` seem very similar, both in description and code. What is their difference, and could they be combined?